AK/BufferedStream.h

```diff
@@ -115,6 +115,73 @@ class BufferedHelper {
         return m_buffer.read(buffer);
     }
 
+    ErrorOr<StringView> read_line_with_resize(ByteBuffer& buffer)
+    {
+        return StringView { TRY(read_until_with_resize(buffer, "\n"sv)) };
+    }
+
+    ErrorOr<Bytes> read_until_with_resize(ByteBuffer& buffer, StringView candidate)
+    {
+        return read_until_any_of_with_resize(buffer, Array { candidate });
+    }
+
+    template<size_t N>
+    ErrorOr<Bytes> read_until_any_of_with_resize(ByteBuffer& buffer, Array<StringView, N> candidates)
+    {
+        if (!stream().is_open())
+            return Error::from_errno(ENOTCONN);
+
+        if (buffer.is_empty())
+            return buffer;
+
+        auto candidate = TRY(find_and_populate_until_any_of(candidates));
+
+        if (candidate.has_value()) {
```

Comment:
The following three cases (`candidate.has_value()`, `m_buffer.used_space() == 0 && stream().is_eof()` and `!stream().is_eof()`) look like they can/should be handled by the logic in/around the while loop. For example, there is no reason why finding the delimiter on the first try should take a separate code path compared to if we had to read some full chunks in between.